### PR TITLE
Parameterized XLMR and Roberta model integration tests

### PR DESCRIPTION
**Pull request by Nayef211**
## Summary
- Refactored model  unit tests to integration tests as a followup from #1491
- Parameterized integration tests to reduce redundant test methods 

## Test
`pytest test/integration_tests/test_models.py`

## Followup
- [ ] Decide the best place to put remaining tests in `test_models.py`
- [ ] Add [`nested_params` helper method](https://fburl.com/xbpvu1sq) from torchaudio to parameterize jit and non-jit model tests

### CODE REVIEW

**Comment by Nayef211** [comment]
@parmeet I just realized #1495 already fixes the truncate transform length issue. I will update my PR to remove this change

File: test/integration_tests/test_models.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import torch
+import torchtext
+
+from ..common.assets import get_asset_path
+from ..common.torchtext_test_case import TorchtextTestCase
+
+
+class TestModels(TorchtextTestCase):
+    def test_xlmr_base(self):
```
Comment (mthrok) [inline test/integration_tests/test_models.py:9]: It seems to me that the underlying logics are same.
You might want to consider parameterizing the tests.

**Comment by parmeet** [comment]
nit: We could potentially split it into two separate PRs- one for unit testing and other for documentation.

**Comment by Nayef211** [comment]
> nit: We could potentially split it into two separate PRs- one for unit/integration testing refactoring and other for documentation.

Sure let me move the documentation to a separate PR to make the changes easier to review

**Comment by Nayef211** [comment]
Here's the PR with just the documentation changes https://github.com/pytorch/text/pull/1497

**Comment by Nayef211** [comment]
@mthrok @parmeet I think this PR is ready for review. Failing dataset tests should be fixed in https://github.com/pytorch/text/pull/1490 and I will rebase this PR on top of it once it's merged

File: test/integration_tests/test_models.py [hunk 2]
```diff
@@ -1,67 +1,56 @@
...
-        model_input = torch.tensor(transform_jit([test_text]))
-        actual = model_jit(model_input)
-        expected = torch.load(asset_path)
-        torch.testing.assert_close(actual, expected)
-
-    def test_roberta_large(self):
-        asset_path = get_asset_path("roberta.large.output.pt")
-        test_text = "Roberta base Model Comparison"
+class TestModels(TorchtextTestCase):
+    @parameterized.expand(TEST_MODELS_PARAMETERIZED_ARGS)
```
Comment (mthrok) [inline test/integration_tests/test_models.py:30]: I think you can go further to make the jit part parameterized as well.

You can parameterize the behavior around jitting in `test_model`, then make a Cartesian product of `TEST_MODELS_PARAMETERIZED_ARGS` and `jit=True|False` to run all the tests.

**Review by mthrok — APPROVED** [review]
Approving as the current state is a good enough improvement. But I highly recommend to go further with parameterization.

File: test/integration_tests/test_models.py (hunk 2 above)
Comment (Nayef211) [reply test/integration_tests/test_models.py:30]: Thanks for the feedback. Will do this in a followup PR by pulling in the `nested_params` helper method from torchaudio.

**Pull request merged**